domain/modeldefaults/state/state.go

```diff
@@ -230,10 +235,276 @@ WHERE m.uuid = ?
 	}, nil
 }
 
-// NewState returns a new State for interacting with the underlying model
-// defaults.
-func NewState(factory database.TxnRunnerFactory) *State {
-	return &State{
-		StateBase: domain.NewStateBase(factory),
+// GetModelCloudDetails returns the cloud UUID and region for the given model.
+// If the model is not found, an error specifying [modelerrors,NotFound] is returned.
+func (s *State) GetModelCloudDetails(ctx context.Context, uuid coremodel.UUID) (cloud.UUID, string, error) {
+	db, err := s.DB()
+	if err != nil {
+		return "", "", errors.Trace(err)
+	}
+
+	model := modelUUID{UUID: uuid.String()}
+	var region cloudRegion
+
+	query, err := s.Prepare(`
+SELECT m.cloud_uuid AS &cloudRegion.cloud_uuid, cr.name AS &cloudRegion.name
+FROM model m
+JOIN cloud_region cr ON cr.uuid = m.cloud_region_uuid
+WHERE m.uuid = $modelUUID.uuid
+`, model, region)
+	if err != nil {
+		return "", "", errors.Trace(err)
+	}
+
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, query, model).Get(&region)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return interrors.Errorf("model %q not found%w", uuid, errors.Hide(modelerrors.NotFound))
+		}
+		return errors.Trace(err)
+	})
+	return cloud.UUID(region.CloudUUID), region.Name, errors.Annotatef(err, "getting cloud details for model %q", uuid)
```

Comment:
We are trying to actively get rid of this sort of thing. Please change this to an `if err != nil`